### PR TITLE
feat: Extend telescope detector with disc surface

### PR DESCRIPTION
**Pull request by XiaocongAi**
This PR extends the telescope detector to be configurable with either `PlaneSurface` or `DiscSurface`. This will allow for reco test with the `DiscSurface`.

### CODE REVIEW

File: Examples/Detectors/TelescopeDetector/include/ActsExamples/TelescopeDetector/BuildTelescopeDetector.hpp [hunk 1]
```diff
@@ -19,22 +19,32 @@
 namespace ActsExamples {
 namespace Telescope {
 
+/// The telescope detector surface type
+enum TelescopeSurfaceType : unsigned int {
```
Comment (paulgessinger) [inline Examples/Detectors/TelescopeDetector/include/ActsExamples/TelescopeDetector/BuildTelescopeDetector.hpp:23]: Do you need the implicit convertability?
```suggestion
enum class TelescopeSurfaceType {
```

File: Examples/Detectors/TelescopeDetector/include/ActsExamples/TelescopeDetector/BuildTelescopeDetector.hpp [hunk 2]
```diff
@@ -19,22 +19,32 @@
...
+/// parallel to
 std::unique_ptr<const Acts::TrackingGeometry> buildDetector(
     const typename TelescopeDetectorElement::ContextType& gctx,
     std::vector<std::shared_ptr<TelescopeDetectorElement>>& detectorStore,
-    const std::vector<double>& positions, std::array<double, 2> offsets,
-    std::array<double, 2> pSize, double thickness,
-    Acts::BinningValue binValue = Acts::BinningValue::binZ);
+    const std::vector<double>& positions, const std::array<double, 2>& offsets,
+    const std::array<double, 2>& bounds, double thickness,
+    const TelescopeSurfaceType& surfaceType,
```
Comment (paulgessinger) [inline Examples/Detectors/TelescopeDetector/include/ActsExamples/TelescopeDetector/BuildTelescopeDetector.hpp:46]:
```suggestion
    TelescopeSurfaceType surfaceType,
```

File: Examples/Detectors/TelescopeDetector/src/BuildTelescopeDetector.cpp [hunk 3]
```diff
@@ -33,14 +35,18 @@ ActsExamples::Telescope::buildDetector(
     std::vector<
         std::shared_ptr<ActsExamples::Telescope::TelescopeDetectorElement>>&
         detectorStore,
-    const std::vector<double>& positions, std::array<double, 2> offsets,
-    std::array<double, 2> pSize, double thickness,
-    Acts::BinningValue binValue) {
+    const std::vector<double>& positions, const std::array<double, 2>& offsets,
+    const std::array<double, 2>& bounds, double thickness,
+    const ActsExamples::Telescope::TelescopeSurfaceType& surfaceType,
+    const Acts::BinningValue& binValue) {
```
Comment (paulgessinger) [inline Examples/Detectors/TelescopeDetector/src/BuildTelescopeDetector.cpp:41]: These probably don't need to be refs.

**Review by paulgessinger — COMMENTED** [review]
Seems fine, I think. I only have a few minor comments.

File: Examples/Detectors/TelescopeDetector/include/ActsExamples/TelescopeDetector/BuildTelescopeDetector.hpp (hunk 1 above)
Comment (XiaocongAi) [reply Examples/Detectors/TelescopeDetector/include/ActsExamples/TelescopeDetector/BuildTelescopeDetector.hpp]: I guess no. Fixed.

File: Examples/Detectors/TelescopeDetector/include/ActsExamples/TelescopeDetector/BuildTelescopeDetector.hpp (hunk 2 above)
Comment (XiaocongAi) [reply Examples/Detectors/TelescopeDetector/include/ActsExamples/TelescopeDetector/BuildTelescopeDetector.hpp]: Fixed.

File: Examples/Detectors/TelescopeDetector/src/BuildTelescopeDetector.cpp (hunk 3 above)
Comment (XiaocongAi) [reply Examples/Detectors/TelescopeDetector/src/BuildTelescopeDetector.cpp]: Fixed.

**Comment by XiaocongAi** [comment]
Hi @paulgessinger , thank you for the suggestions. I have made the update.

**Review by paulgessinger — APPROVED** [review]
Ok then.

**Review by asalzburger — APPROVED** [review]

**Pull request merged**